manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -282,6 +289,32 @@ func (tr *TaskReaper) tick() {
 			return nil
 		})
 	}
+
+	// Perform service cleanup
+	deleteServices := make(map[string]struct{})
```

Comment:
Should the trigger for service cleanup instead be task status update event ?

Reply:
I think it's okay to do cleanups on ticks, rather than every task update. For a service that has a large number of tasks, maybe it's too much unnecessary actions we'd be taking.

Reply:
Makes sense